File: BaekJoon/5622_다이얼.cpp
#include <iostream>
#include <string>

#define ARR_SIZE 9
#define BASIC_TIME_TAKEN 2

int main(void)
{
    std::string dial[ARR_SIZE] = { "", "ABC", "DEF", "GHI", "JKL", "MNO", "PQRS", "TUV", "WXYZ" };
    int result = 0;
    std::string data;
    
    std::cin >> data;
    for(int i=0; i<data.length(); i++) {
        for(int j=1; j<ARR_SIZE; j++) {
            if(dial[j].find(data[i]) != std::string::npos) {
                result += (BASIC_TIME_TAKEN+j);
                break;   
            }
        }
    }

    std::cout << result << '\n';
    return 0;
}

File: 01-baekjoon/5622_다이얼.swift
import Foundation

// 다이얼 전화기의 문자(Key)와 해당 숫자를 거는 데 걸리는 시간(Value)
let dials = ["ABC": 3, "DEF": 4, "GHI": 5, "JKL": 6,
            "MNO": 7, "PQRS": 8, "TUV": 9, "WXYZ": 10]

// 문자열 S 입력 받기
let alpha = readLine()!
// 문자열을 순회하며 다이얼에 해당하는 키값 검색 및 시간 계산하기
var result = 0
for c in alpha {
    let s = String(c)
    for dial in dials where dial.key.contains(s) {
        result += dial.value
    }
}
// 결과 답안 출력
print(result)
